### DISCUSSION TITLE
Why does my web server not show the 'error.html' page?

### DISCUSSION THREAD

**Discussion by hysonlee** [datenstrom/community#116] [Ask a question] [answered]
Hi there,
I built my site into static page(s), visit the site is normally OK, but when occur something missing (404 error), it not display the `error.html` page and just echo **"File not found."**. I'm sure the `error.html` file is exist and pretty fine.

My website holding with a Linux+Apache hosting, the Yellow installation I used to produce static site is up to date.

**Comment by markseuffert** [comment]
Hi, can you show your `[debug server]` settings and the content of your `.htaccess` file? Thanks.

**Comment by hysonlee** [comment]
Here is my `debug server`:

```
ServerScheme: http
ServerName: www.domain.com
ServerBase:
ServerTime: UTC+8
```

and here is the `.HTACCESS`content:

```
<IfModule mod_rewrite.c>
RewriteEngine on

# Yellow dynamic pages

RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^ yellow.php [L]
DirectoryIndex index.html yellow.php
RewriteRule ^(cache|content|system)/ error [L]

# Yellow static pages

RewriteCond %{REQUEST_FILENAME} -f
RewriteRule ^[^\.]+$ - [T=text/html,L]
ErrorDocument 404 /error.html
</IfModule>
```

**Comment by hysonlee** [comment]
Also I encountered a strange thing: one of my page within **footnote**, after static translation, folders with the same amount of footnotes be created at the same directory of static base, each footnote folder contain one `index.html`file.

```
 cache#fn:????????????
 cache#fnref:AS
 cache#fn:AS        
 cache#fnref:AS4538info
 cache#fn:AS4538info    
 cache#fnref:AS???????????????
 cache#fn:AS???????????????
 cache#fnref:????????????
```

**Comment by markseuffert** [comment]
Your settings look good. I wasn't able to reproduce the problem, anyone else has the same problem?

**Comment by hysonlee** [comment]
I can't believe my web hosting works fine with Yellow installation but problem with static site.

**Comment by markseuffert** [comment, marked as answer]
It took me some time to figure this out. The web hoster is http://gandi.net and you need to modify your `.htaccess` file. If you remove the Yellow dynamic rules, they are not needed for static pages anyway, then will it show the custom error page. The strange thing is that this web hoster runs Apache 2.4, which is used by the Yellow demo without having the same problem. 

Here's an example:

```
<IfModule mod_rewrite.c>
RewriteEngine on

# Yellow dynamic pages

# RewriteCond %{REQUEST_FILENAME} !-f
# RewriteCond %{REQUEST_FILENAME} !-d
# RewriteRule ^ yellow.php [L]
# DirectoryIndex index.html yellow.php
# RewriteRule ^(cache|content|system)/ error [L]

# Yellow static pages

RewriteCond %{REQUEST_FILENAME} -f
RewriteRule ^[^\.]+$ - [T=text/html,L]
ErrorDocument 404 /error.html
</IfModule>
```

I don't know why this is necessary, but it seams to do the trick. Hope this helps. :smile:

**Comment by hysonlee** [comment]
Mark, Thank you for your hard troubleshooting, I think I'd start a ticket to the Gandi support team for this.

**Comment by markseuffert** [comment]
See also #125
